ast.c

```diff
@@ -850,6 +870,19 @@ static void ast_fill_children_ht(HashTable *ht, zend_ast *ast, ast_state_info_t
 	}
 #endif
 
+#if PHP_VERSION_ID < 80400
+	if (state->version >= 110) {
+		if (ast_kind == ZEND_AST_PARAM || ast_kind == ZEND_AST_PROP_ELEM) {
+			zval tmp;
+			ZVAL_NULL(&tmp);
+			zend_hash_add_new(ht, AST_STR(str_hooks), &tmp);
+			/* For now, ast_kind_is_decl is false. */
+			/* TODO a future backwards incompatible release dropping support for older AST versions might have property elements as declarations? */
```

Comment:
I don't get this comment. ast_kind_is_decl() is determined by PHP, not this extension. It's about whether it uses zend_ast_decl.

Reply:
Correct - I think I mixed up node kinds here, in retrospect - neither of those use `zend_ast_decl` - I was mixing those up with nodes that had reflection in general